app/models/request.rb

```diff
@@ -169,15 +170,23 @@ def library_id_error?
     errors[:library_id].present?
   end
 
+  def pickup_libraries
+    location_rule&.pickup_libraries
+  end
+
   class << self
     # The mediateable_origins will make multiple (efficient) database requests
     # in order to return the array of locations that are both configured as mediateable and have existing requests.
     # Another alternative would be to use (origin_admin_groups & uniq.pluck(:origin)).present? but that will result
     # in a SELECT DISTINCT which could get un-performant with a large table of requests.
     def mediateable_origins
-      Settings.origin_admin_groups.to_hash.keys.map(&:to_s).select do |library_or_location|
-        MediatedPage.exists?(origin: library_or_location) || MediatedPage.exists?(origin_location: library_or_location)
-      end
+      Settings.mediateable_origins.map.select do |code, config|
+        if config.library_override
+          MediatedPage.exists?(origin_location: code.to_s)
+        else
+          MediatedPage.exists?(origin: code.to_s)
+        end
+      end.to_h.stringify_keys
```

Comment:
Can you say more about the need for string-key hashes? Path of least resistance?

Reply:
Yup. We had string keys before, tests that said they should be strings, and stuff that broke if they weren't 🤷‍♂️

Reply:
Meh, easy enough to stick with string keys.